src/main/java/com/nylas/NeuralCleanConversation.java

```diff
@@ -0,0 +1,53 @@
+package com.nylas;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class NeuralCleanConversation extends Message {
+	private String conversation;
+	private String model_version;
+
+	public String getConversation() {
+		return conversation;
+	}
+
+	public void setConversation(String conversation) {
+		this.conversation = conversation;
+	}
+
+	public String getModelVersion() {
+		return model_version;
+	}
+
+	public void setModelVersion(String modelVersion) {
+		this.model_version = modelVersion;
+	}
+
+	public List<File> extractImages(NylasAccount account) throws RequestFailedException, IOException {
+		List<File> fileList = new ArrayList<>();
+		if(conversation != null) {
+			// After applying the regex, if there are IDs found they would be
+			// in the form of => 'cid:xxxx' (including apostrophes), so we discard
+			// everything before and after the file ID (denoted as xxxx above)
+			Matcher fileIdMatcher = Pattern.compile("[(']cid:(.)*[)']").matcher(conversation);
+			while (fileIdMatcher.find()) {
+				String match = fileIdMatcher.group();
+				String fileId = match.substring(5, match.length() - 1);
+				fileList.add(account.files().get(fileId));
```

Comment:
Currently I believe we strictly separate the API calls to data access objects, so I would prefer we continue that.  What do you think of moving this method into the `Neural` or `Files` classes and pass this `NeuralCleanConversation` in?

Reply:
Hmmm.... that's a fair point. Let me investigate putting it in `Neural`. Thanks!